plugin/storage/integration/elasticsearch_test.go

```diff
@@ -249,3 +254,32 @@ func (s *ESStorageIntegration) cleanESIndexTemplates(t *testing.T, prefix string
 	}
 	return nil
 }
+
+// testArchiveTrace validates that a trace with a start time older than maxSpanAge
+// can still be retrieved via the archive storage. This ensures archived traces are
+// accessible even when their age exceeds the retention period for primary storage.
+// This test applies only to Elasticsearch (ES) storage.
+func (s *ESStorageIntegration) testArchiveTrace(t *testing.T) {
+	s.skipIfNeeded(t)
+	defer s.cleanUp(t)
+	tID := model.NewTraceID(uint64(11), uint64(22))
+	expected := &model.Span{
+		OperationName: "archive_span",
+		StartTime:     time.Now().Add(-maxSpanAge * 5).Truncate(time.Microsecond),
```

Comment:
maxSpanAge is a constant in this file, but it is not passed as a configuration to the storage factory

Reply:
let's add in a separate PR

Reply:
@yurishkuro its just using the default - https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/es/options.go#L403

Reply:
I know, but it's disassociated from that default. We should pass max age as parameter to the factory here.